sources/org.osbuild.curl

```diff
@@ -83,55 +83,168 @@ SCHEMA = """
 """
 
 
-def fetch(url, checksum, directory):
+class CurlResult(SourceResult):
+
+    INIT = 1
+    ERROR_DOWNLOADING = 1
+    ERROR_CHECKSUM = 2
+    DONE = 3
+
+    def __init__(self, source: Dict):
+        super().__init__(source)
+        self.state = CurlResult.INIT
+
+    def get_error_message(self) -> str:
+        if self.state == CurlResult.INIT:
+            return "unknown"
+        if self.state == CurlResult.ERROR_CHECKSUM:
+            return "Checksum mismatch"
+        if self.state == CurlResult.ERROR_DOWNLOADING:
+            return f"{CurlResult.CURL_CODE_MAP.get(self.return_code, 'unknown')}"
+        return None
+
+    CURL_CODE_MAP = {1: "Unsupported protocol",
```

Comment:
I am not sure we should report all these errors back. We might want to segment them into programming errors (like 3 "url malformed", or 22 "quote error"), network errors and maybe security relevant errors (91 "invalid ssl cert", checksumming failed) the rest.

Reply:
Alright, I'm doing this.

I defined this errors:
```
    CURL_BAD_INPUT
    CURL_INTERNAL_ERROR
    CURL_NETWORKING
    CURL_CREDENTIALS
    CURL_STORAGE
    CURL_SECURITY
```

I think, we can add one other CURL_UNSUPPORTED for whatever we protocol we do not want to add support (thinking of FTP or LDAP maybe ?)

Reply:
However I kept all of the errors for now, I think we need to decide how we reduce that.